woocommerce/woocommerce
# Add missing tracks events to product editor

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds tracks discussed in the issue as well as missing properties to fill in missing gaps in tracks data.  Also creates a `source` property to better identify where events are triggered from.

@pmcpinto made the following changes to the events suggested:

* `product_add_description_button` -> `product_add_description_click`
* `product_add_attribute_button ` -> `product_add_attributes_click` (note I made the assumption that this event is fired when the "Add attributes" button is clicked to open the attributes modal)
* `product_dropdown_option` -> `product_dropdown_option_click` (options: `feedback` | `classic_editor`)

Closes #38637 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce -> Settings -> Advanced -> Features and enable the new product editing experience
2. Open your console and enter `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
3. Navigate to "Add products"
4. Attempt to run all the events noted in #38637 
5. Make sure those events run in the console.  Note that `wcadmin_product_add_publish` will fire on the backend and can be found under WooCommerce -> Status -> Logs and finding the latest `tracks-2023-xx-xx` logs.
